codazen/webDiplomacy
# Initial commit that does polling based on user activity.

### Summary
This PR adds polling capabilities to the game, updating the game data if it has progressed to the next phase. To know whether or not the phase has progressed, it compares the following data from the `game/overview` route:

1. season
2. year
3. processTime

If a change in any of these values is detected, it means that the game has progressed and it makes a call to fetch new game data. The new data creates a cascade effect which updates the map as well as information in the UI.

###Video
In the video below, polling is happening **at most** once every 30 seconds, based on user activity. When the game is progressed (done by readying in the legacy app) the poll detects that the game has progressed and makes call to update the game data. Once the game data is updated, the map and UI updates itself. This can be seen as the fleet in the Baltic Sea is redrawn in Prussia, and the top left season icon is updated from Spring to Autumn. 

https://drive.google.com/file/d/1Er4ZQHp6H78KyBGxk8deYLA-K2bO6wDG/view?usp=sharing

###Testing Plan

1. Load a game in your local environment
2. Check the network tab to verify that polling is happening at most every 30 seconds (based on user activity)
3. Ready and progress turns on the legacy app, **do not reload the beta app**, then verify that the beta app detects a game progression and updates the game data / map / UI accordingly.
